modules/compute-vm/resource-policies.tf

```diff
@@ -118,22 +141,25 @@ resource "google_compute_resource_policy" "snapshot" {
 }
 
 resource "google_compute_disk_resource_policy_attachment" "boot" {
-  count   = var.boot_disk.snapshot_schedule != null ? 1 : 0
-  project = var.project_id
-  zone    = var.zone
+  for_each = var.boot_disk.snapshot_schedule != null ? {
+    for i, v in var.boot_disk.snapshot_schedule : i => v
```

Comment:
nit: can you just use the variable instead of a `for` loop that returns the same?